src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature

```diff
@@ -45,10 +45,12 @@ Feature: Clinical Visit Management
     When a user clicks on end visit button
     Then the system ends the patient visit
   
+  @selenium
+  @clinicalVisit
   Scenario: Clinical Visit for uploading supporting file
```

Comment:
why not populating the user story on the same patient?

Reply:
Oh yeah @kdaud , that is it, let me make the test to work with the same patient.

Reply:
@kdaud  though something funny is still happening on my side. While using firefox as opposed to chrome, the action to end a visit does not seem to work. The test still shows that the visit is not ended so it throws an error here 
` @Then("the system ends the patient visit")
	public void systemEndsPatientVisit() {
		assertNull(visitsDashboardPage.getActiveVisit());
		dashboardPage = visitsDashboardPage.goToPatientDashboard();
	}`
assertNull fails. So I tried to test it by putting assertNotNull and the test passed on firefox meaning that the visit was still active. However, with chrome, the test passes normally.

Reply:
> 

The user story needs to be invoked before ending the visit. This would mean inserting it before the end visit steps [steps](https://github.com/openmrs/openmrs-contrib-qaframework/blob/e59864394a6bc8fb9913117da5036c5ffca8738e/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature#L42)

Reply:
> 

Ah ok. let me try that @kdaud !

Reply:
> the action to end a visit does not seem to work. The test still shows that the visit is not ended so it throws an error here

@Irenyak1 the step is passing well via ci but if its failing locally then you can log a ticket here and resolve it so that the workflow test is stable both locally and via ci.

Reply:
Ok @kdaud , let me first try altering the steps and I see if it will pass. If it fails, I will log a ticket. Thanks.

Reply:
@kdaud  I think I am going to log a ticket because the test still shows the error I mentioned earlier, it does not end the visit. i.e 

` 1638783553144	Marionette	INFO	Stopped listening on port 41809
console.warn: TopSitesFeed: Failed to fetch data from Contile server: NetworkError when attempting to fetch resource.
console.error: Region.jsm: "Error fetching region" (new TypeError("NetworkError when attempting to fetch resource.", ""))
console.error: Region.jsm: "Failed to fetch region" (new Error("NO_RESULT", "resource://gre/modules/Region.jsm", 419))
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 141.023 sec <<< FAILURE!
Clinical Visit(Clinical Visit Management)  Time elapsed: 140.336 sec  <<< FAILURE!
java.lang.AssertionError: expected null, but was:<[[FirefoxDriver: firefox on LINUX (ca38b5a8-b149-4791-a7f8-52f2fdc2fb88)] -> css selector: #visits-list li.menu-item.viewVisitDetails span.menu-date]>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotNull(Assert.java:756)
	at org.junit.Assert.assertNull(Assert.java:738)
	at org.junit.Assert.assertNull(Assert.java:748)
	at org.openmrs.contrib.qaframework.automation.ClinicalVisitSteps.systemEndsPatientVisit(ClinicalVisitSteps.java:293)
	at ✽.the system ends the patient visit(file:///media/irynah/DATA/0PROJECTS/OpenMRS-projects/openmrs-contrib-qaframework/src/test/resources/features/refapp-2.x/14-clinicalVisit/clinicalVisit.feature:52)

`

Reply:
@Irenyak1 could you update the PR with your changes and the reviewers take a look at the logs via ci  !!

Reply:
@kdaud  can I do a surgery on this work flow. I shift the user stories in the feature file to match step definition file. Is that ok, or I leave it as it is? **Note** What comes first in the feature file is not what comes first in the step definition file. I know it is not affecting the flow of the test but just my thoughts.

Reply:
> 

Sure @kdaud  could you reply to the request I have just made so I can push the changes.

Reply:
> I know it is not affecting the flow of the test

I would focus on what is crucial for the test to pass the run.

Reply:
Ok. let me push my changes then.

Reply:
@kdaud  I have pushed the changes, you may review them.

Reply:
Hi @kdaud, please could you look are this [ticket ](https://issues.openmrs.org/browse/RATEST-278) and curate it if possible. Thanks.